storage/innobase/xtrabackup/doc/source/xbcloud/xbcloud_exbackoff.rst

```diff
@@ -0,0 +1,104 @@
+
+The operation is retried until the ``--max-retries`` value is reached. If the chunk operation fails on the last retry, xbcloud aborts the process.
+
+The default values are the following:
+
+* --max-backoff = 300000 (5 minutes)
+
+* --max-retries = 10
+
+
+You can adjust the number of retries by adding the ``--max-retries`` parameter and adjust the maximum length of time between retries by adding the ``--max-backoff`` parameter to an xbcloud command. 
+
+Since xbcloud does multiple asynchronous requests in parallel, a calculated value, measured in milliseconds, is used for ``max-backoff``. This algorithm calculates how many milliseconds to sleep before the next retry. A number generated is based on the combining the power of two (2), the number of retries already attempted and adds a random number between 1 and 1000. This number avoids network congestion if multiple chunks have the same backoff value. If the default values are used, the final retry attempt to be approximately 17 minutes after the first try. The number is no longer calculated when the milliseconds reach the ``--max-backoff`` setting. At that point, the retries continue by using the ``--max-backoff`` setting until the ``max-retries`` parameter is reached.
+
+.. _retriable:
+
+Retriable errors
+------------------
+
+We retry for the following CURL operations:
+
+* CURLE_GOT_NOTHING
+
+* CURLE_OPERATION_TIMEOUT
+
+* CURLE_RECV_ERROR
+
+* CURLE_SEND_ERROR
+
+* CURLE_SEND_FAIL_REWIND
+
+* CURLE_PARTIAL_FILE
+
+* CURLE_SSL_CONNECT_ERROR
+
+We retry for the following HTTP operation status codes:
+
+* 503
+
+* 500
+
+* 504
+
+* 408
+
+Each cloud provider may return a different CURL error or an HTTP error, depending on the issue. Add new errors by setting the following variables ``--curl-retriable-errors`` or ``--http-retriable-errors`` on the command line or in ``my.cnf`` or in a custom configuration file under the [xbcloud] section.
+
+The error handling is enhanced when using the ``--verbose`` output. This output specifies which error caused xbcloud to fail and what parameter a user must add to retry on this error. 
+
+The following is an example of a verbose output:
+
+.. sourcecode:: bash
+
+    210701 14:34:23 /work/pxb/ins/2.4/bin/xbcloud: Operation failed. Error: Server returned nothing (no headers, no data)
+    210701 14:34:23 /work/pxb/ins/2.4/bin/xbcloud: Curl error (52) Server returned nothing (no headers, no data) is not configured as retriable. You can allow it by adding --curl-retriable-errors=52 parameter
+
+Example
+--------
+The following example adjusts the maximum number of retries and the maximum time between retries.
+
+.. sourcecode:: bash
+
+    xtrabackup --backup --stream=xbstream --target-dir=/tmp --max-retries=5 --max-backoff=10000
```

Comment:
hmm, this is not correct, we should adjust on 8.0 too.

```suggestion
    xbcloud [options] --max-retries=5 --max-backoff=10000
```

Reply:
Updated.